k8s/upgrade/src/bin/upgrade-job/upgrade/utils.rs

```diff
@@ -32,27 +33,42 @@ pub(crate) async fn rebuild_result(
         }
 
         for volume in unhealthy_volumes.iter() {
-            match replica_rebuild_count(volume.clone()).await {
-                0 => {
-                    for _i in 0 .. 11 {
-                        // wait for a minute for any rebuild to start
-                        tokio::time::sleep(Duration::from_secs(60_u64)).await;
-                        let count = replica_rebuild_count(volume.clone()).await;
-                        if count > 0 {
+            if let Some(target) = volume.state.target.as_ref() {
+                let mut volume_over_nodes = HashSet::new();
+                volume_over_nodes.insert(target.node.as_str());
+
+                for (_, topology) in volume.state.replica_topology.iter() {
+                    if let Some(node) = topology.node.as_ref() {
+                        volume_over_nodes.insert(node);
+                    }
+                }
+
+                if volume_over_nodes.contains(node_name) {
+                    match replica_rebuild_count(volume.clone()).await {
+                        0 => {
+                            for _i in 0 .. 11 {
+                                // wait for a minute for any rebuild to start
+                                tokio::time::sleep(Duration::from_secs(60_u64)).await;
+                                let count = replica_rebuild_count(volume.clone()).await;
```

Comment:
Existing code but since we're here... seems `replica_rebuild_count` doesn't need the value, we could pass reference instead and also doesn't need to be async.

Reply:
done